src/goto-symex/symex_main.cpp

```diff
@@ -54,6 +93,13 @@ void goto_symext::claim(const expr2tc &claim_expr, const std::string &msg)
   if(is_true(new_expr))
     return;
 
+  if(options.get_bool_option("smt-symex-assert"))
+  {
+    if(check_incremental(new_expr, true, msg))
+      // incremental verification has succeeded
+      return;
+  }
+
```

Comment:
Shouldn't we add the result of the evaluation to the guards if the assertion evaluates to false?

Reply:
Fixed.